SplitFiles/control.c

```diff
@@ -51,13 +49,13 @@ void buffer_restack(const ControlBuffer* const buff) {
     char* command;
     const int buffer_index = buff->bufferIndex;
     // first clean the buffer
-    for (i = 2; i <= CONTROL_BUFFER_SIZE; i++) {
-        printf(PRINTAT"%c%c""            \0", 21, i);
+    for (i = 1; i <= CONTROL_BUFFER_SIZE; i++) {
+        printf(PRINTAT"%c%c""   \0", 28, i);
     }
     // now write out stack
-    for (i = 2; i <= buffer_index; i++) {
+    for (i = 1; i <= buffer_index; i++) {
         command = buffer_getcommand((buff->buffer[i - 1]));
-        printf(PRINTAT"%c%c""%s\0", 21, i, command);
+        printf(PRINTAT"%c%c""%s\0", 28, i, command);
     }
```

Comment:
Quicker to overwrite the stack only once:
for (i = 0; i < buffer_index; ++i) {
    //Write out stack, using fixed-width formatting
    ...
}
for (; i < CONTROL_BUFFER _SIZE; ++i) {
    //Write out spaces, also using fixed-width formatting
    ...
}

the (; above just allows you to use i as it was from the end of the prior loop

Fixed-width formatting looks like %-12s for 12 chars wide

Reply:
This doesn't work! It prints all the empty slots as "EH?" and then drops the top slot